dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -67,392 +67,345 @@
         }
-
-        AdminLogger.log(ESIndexResource.class, "activateIndex", "Index activated: " + indexName);
+        return Response.ok(new ResponseEntityView(builder.build())).build();
     }
-
-    public static void deactivateIndex(String indexName) throws DotDataException, IOException {
-    	AdminLogger.log(ESIndexResource.class, "deactivateIndex", "Trying to deactivate index: " + indexName);
-
-        if(indexName.startsWith(SiteSearchAPI.ES_SITE_SEARCH_NAME)){
-            APILocator.getSiteSearchAPI().deactivateIndex(indexName);
-        }
-        else{
-            APILocator.getContentletIndexAPI().deactivateIndex(indexName);
-        }
-
-        AdminLogger.log(ESIndexResource.class, "deactivateIndex", "Index deactivated: " + indexName);
+    
+    @CloseDBIfOpened
+    @GET
+    @JSONP
+    @NoCache
+    @Path("/failed")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response downloadRemainingRecordsAsCsv(@Context final HttpServletRequest request,
+                    @Context final HttpServletResponse response) throws DotDataException {
+        auth(request, response);
+        List<Map<String, Object>> results = new ArrayList<>();
+        
+        
+        
+        APILocator.getReindexQueueAPI().getFailedReindexRecords().stream().forEach(row->{
+            Map<String, Object> failure = new HashMap<>();
+            
+            final Contentlet contentlet = Try.of(()->APILocator.getContentletAPI().findContentletByIdentifierAnyLanguage(row.getIdentToIndex()))
+                            .onFailure(e->Logger.warn(ESIndexResource.class, e.getMessage(), e))
+                            .getOrElse(Contentlet::new);
+            Map<String, Object> conMap = Try.of(() -> new ContentletToMapTransformer(contentlet).toMaps().get(0))
+                            .onFailure(e->Logger.warn(ESIndexResource.class, e.getMessage(), e))
+                            .getOrElse(HashMap::new);
+            
+            failure.put("title", contentlet.getTitle());
+            failure.put("inode", contentlet.getInode());
+            failure.put("identifier", row.getIdentToIndex());
+            failure.put("serverId", row.getServerId());
+            failure.put("failureReason", row.getLastResult());
+            failure.put("priority", row.getPriority());
+            failure.put("contentlet", conMap);
+            results.add(failure);
+        });
+
+        return Response.ok(results).build();
     }
 
-    public static long indexDocumentCount(final String indexName) {
-        return APILocator.getContentletIndexAPI().getIndexDocumentCount(indexName);
+    @CloseDBIfOpened
+    @DELETE
+    @JSONP
+    @NoCache
+    @Path("/failed")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response deleteFailedRecords(@Context final HttpServletRequest request, @Context final HttpServletResponse response)
+                    throws DotDataException {
+        final InitDataObject init = auth(request, response);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)